ams/optimizer/src/main/java/com/netease/arctic/optimizer/LocalOptimizerContainer.java

```diff
@@ -40,7 +44,8 @@ protected Map<String, String> doScaleOut(String startUpArgs) {
     try {
       String[] cmd = {"/bin/sh", "-c", startUpArgs};
       LOG.info("Starting local optimizer using command:" + startUpArgs);
-      runtime.exec(cmd);
+      Process process = runtime.exec(cmd);
```

Comment:
The process status should be checked first. You can use process.waitFor() to wait for the program to complete execution and obtain the exit status code.

Reply:
I have tried using `process.waitFor()` to retrieve the exit status code and check the process status, but I encountered the following two issues:

1. When `process.waitFor()` returns, I am unable to retrieve the information from the input stream in `process.getErrorStream()` because the input stream is destroyed by the time `process.waitFor()` returns.

2. The `process.waitFor()` method returns the operating system error code. For example, a status code of 127 indicates insufficient permissions to execute the file. However, if there is an internal error within the executed file, such as the inability to find the main class due to a missing lib, `process.waitFor()` will still return 0. This means that I cannot directly determine whether the process was successfully created based solely on the exit status code.

Considering these two issues, I believe that `process.waitFor()` is not suitable for the current scenario.

Reply:
You need read input in anther thread，The following code is from Hadoop, you can refer to it.
```
  public int run(List<String> command, List<String> output) {
    int retCode = 1;
    ProcessBuilder pb = new ProcessBuilder(command);
    try {
      Process p = pb.start();
      OutputBufferThread stdOut = new OutputBufferThread(p.getInputStream());
      OutputBufferThread stdErr = new OutputBufferThread(p.getErrorStream());
      stdOut.start();
      stdErr.start();
      retCode = p.waitFor();
      if (retCode != 0) {
        mojo.getLog().warn(command + " failed with error code " + retCode);
        for (String s : stdErr.getOutput()) {
          mojo.getLog().debug(s);
        }
      }
      stdOut.join();
      stdErr.join();
      output.addAll(stdOut.getOutput());
    } catch (Exception ex) {
      mojo.getLog().warn(command + " failed: " + ex.toString());
    }
    return retCode;
  }
```

Reply:
Thank you for your suggestion. I will refer to the Hadoop code.